### PR TITLE
ATO-319: New GitHub Action to Trigger Orch Deployment in Parallel

### PR DESCRIPTION
**Pull request by gn-GDS**
## What?

New GitHub action that builds out the Java artefact and invokes the Dev Platform action for secure pipelines.

## Why?

Require for phase 3 of orch/auth split.

## Related PRs

[ATO-307](https://govukverify.atlassian.net/jira/software/c/projects/ATO/boards/390?selectedIssue=ATO-307)

[ATO-307]: https://govukverify.atlassian.net/browse/ATO-307?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/deploy-orch.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: Deploy Orchestration
+
+env:
+  AWS_REGION: eu-west-2
+
+#gha role is build Auth github actions role in Gds dev Account
+#Artifact bucket is Build Auth deploy github artifcat Source bucket
+#Destination Bucket is auth-lambda-signed bucket from prod tooling account
```
Comment (ethanmills) [inline .github/workflows/deploy-orch.yml:8]: I don't really understand this comment

Comment (gn-GDS) [reply .github/workflows/deploy-orch.yml:8]: It was copied from auth, will remove 👍

File: .github/workflows/deploy-orch.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: gradle
+
+      - name: Set up Gradle
+        uses: gradle/gradle-build-action@v2
```
Comment (ethanmills) [inline .github/workflows/deploy-orch.yml:36]: What do we need these for?

File: .github/workflows/pre-merge-checks-sam.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+name: Pre-merge checks for Terraform
```
Comment (ethanmills) [inline .github/workflows/pre-merge-checks-sam.yml:1]: Oh no it isn't

File: template.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        Variables:
+          # checkov:skip=CKV_AWS_173: These environment variables do not require encryption.
+          OIDC_API_BASE_URL: !Sub https://oidc.${Environment}.account.gov.uk/
+          ENVIRONMENT: !Sub ${Environment}
+          FRONTEND_BASE_URL: !Sub https://signin.${Environment}.account.gov.uk/
+      Events:
+        WellKnown:
+          Type: Api
+          Properties:
+            Path: /Wellknown
```
Comment (ethanmills) [inline template.yaml:53]: Don't think this is quite right? Or is this just for testing?

Comment (gn-GDS) [reply template.yaml:53]: This is just for testing, it will be updated as part of ATO-320

File: .github/workflows/deploy-orch.yml (hunk 2 above)
Comment (gn-GDS) [reply .github/workflows/deploy-orch.yml:36]: @ethanmills isn't this required for SAM build/deploy?
Its used in other github deployments:
[life events](https://github.com/govuk-one-login/life-events-platform/blob/main/.github/workflows/application-main.yml)
[demo app](https://github.com/govuk-one-login/devplatform-demo-sam-app/blob/main/.github/workflows/sam-app-deploy-using-environment-secrets.yml)

Comment (ethanmills) [reply .github/workflows/deploy-orch.yml:32]: Those also build the Java, here it looks like we're outsourcing that to `Build modules`

Comment (gn-GDS) [reply .github/workflows/deploy-orch.yml:32]: Ahh yep different pipeline structure 👍

File: template.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            Method: get
+      Tags:
+        CheckovRulesToSkip: CKV_AWS_116.CKV_AWS_117.CKV_AWS_173
```
[the remark this answers is not in the document]
Comment (ethanmills) [reply template.yaml:56]: Assuming we're going to handle all this in a followup?

**Review by ethanmills — APPROVED** [review]

File: template.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (gn-GDS) [reply template.yaml:56]: Yep this is a very rough first draft just to check that it validates and deploys. ATO-320 will focus on the template 👍

**Pull request merged**